connexion/apps/async_app.py

```diff
@@ -0,0 +1,268 @@
+"""
+This module defines a native connexion asynchronous application.
+"""
+
+import asyncio
+import logging
+import pathlib
+import pkgutil
+import sys
+import typing as t
+
+from starlette.responses import JSONResponse as StarletteJSONResponse
+from starlette.responses import Response as StarletteResponse
+from starlette.routing import Router
+from starlette.types import Receive, Scope, Send
+
+from connexion.apis.abstract import AbstractAPI
+from connexion.apps.abstract import AbstractApp
+from connexion.decorators.lifecycle import RequestResponseDecorator
+from connexion.decorators.parameter import parameter_to_arg
+from connexion.exceptions import MissingMiddleware, ProblemException
+from connexion.lifecycle import ConnexionResponse, MiddlewareRequest, MiddlewareResponse
+from connexion.middleware.main import ConnexionMiddleware
+from connexion.middleware.routing import ROUTING_CONTEXT
+from connexion.operations import AbstractOperation
+from connexion.uri_parsing import AbstractURIParser
+
+logger = logging.getLogger("Connexion.app")
+
+
+class AsyncAsgiApp:
+    """Mixin for usage with Abstract Apps & Apis"""
+
+    def __init__(self, *args, base_path="", **kwargs):
+        self.apis: t.Dict[str, AsyncApi] = {}
+        self.operations: t.Dict[str, AsyncOperation] = {}
+        self.router = Router()
```

Comment:
Add some documentation on this class about how the routing works (RoutingMiddleware + Router)